src/audio/mixer/mixer.c

```diff
@@ -40,195 +39,87 @@ DECLARE_SOF_RT_UUID("mixer", mixer_uuid, 0xbc06c037, 0x12aa, 0x417c,
-						      dev->frames);
-	buffer_release(sink_c);
-
-	if (sink_period_bytes == 0) {
-		comp_err(dev, "mixer_params(): period_bytes = 0");
-		return -EINVAL;
-	}
-
-	if (sink_stream_size < sink_period_bytes) {
-		comp_err(dev, "mixer_params(): sink buffer size %d is insufficient < %d",
-			 sink_stream_size, sink_period_bytes);
-		return -ENOMEM;
-	}
-
-	return 0;
-}
-
-/* used to pass standard and bespoke commands (with data) to component */
-static int mixer_trigger_common(struct comp_dev *dev, int cmd)
-{
-	int ret;
-
-	ret = comp_set_state(dev, cmd);
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		return PPL_STATUS_PATH_STOP;
-
-	return ret;
-}
 
 /*
  * Mix N source PCM streams to one sink PCM stream. Frames copied is constant.
  */
-static int mixer_copy(struct comp_dev *dev)
+static int mixer_process(struct processing_module *mod,
+			 struct input_stream_buffer *input_buffers, int num_input_buffers,
+			 struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct mixer_data *md = comp_get_drvdata(dev);
-	struct comp_buffer *source, *sink;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
-	struct comp_buffer *sources[PLATFORM_MAX_STREAMS];
-	struct comp_buffer __sparse_cache *sources_c[PLATFORM_MAX_STREAMS];
+	struct mixer_data *md = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	const struct audio_stream __sparse_cache *sources_stream[PLATFORM_MAX_STREAMS];
-	struct list_item *blist;
 	int32_t i = 0;
-	int32_t num_mix_sources = 0;
 	uint32_t frames = INT32_MAX;
 	/* Redundant, but helps the compiler */
 	uint32_t source_bytes = 0;
 	uint32_t sink_bytes;
 
-	comp_dbg(dev, "mixer_copy()");
+	comp_info(dev, "mixer_copy() %d", num_input_buffers);
 
-	/* calculate the highest runtime component status
-	 * between input streams
-	 */
-	list_for_item(blist, &dev->bsource_list) {
-		source = container_of(blist, struct comp_buffer, sink_list);
-		source_c = buffer_acquire(source);
-
-		/* only mix the sources with the same state with mixer */
-		if (source_c->source && source_c->source->state == dev->state)
-			sources[num_mix_sources++] = source;
-
-		buffer_release(source_c);
-
-		/* too many sources ? */
-		if (num_mix_sources == PLATFORM_MAX_STREAMS - 1)
-			return 0;
-	}
-
-	sink = list_first_item(&dev->bsink_list, struct comp_buffer,
-			       source_list);
-	sink_c = buffer_acquire(sink);
+	/* too many sources ? */
+	if (num_input_buffers == PLATFORM_MAX_STREAMS - 1)
+		return 0;
 
 	/* check for underruns */
-	for (i = 0; i < num_mix_sources; i++) {
+	for (i = 0; i < num_input_buffers; i++) {
 		uint32_t avail_frames;
 
-		source_c = buffer_acquire(sources[i]);
-		avail_frames = audio_stream_avail_frames(&source_c->stream,
-							 &sink_c->stream);
+		avail_frames = audio_stream_avail_frames(mod->input_buffers[i].data,
+							 mod->output_buffers[0].data);
 		frames = MIN(frames, avail_frames);
 
-		if (i == num_mix_sources - 1)
+		if (i == num_input_buffers - 1)
```

Comment:
Can this if be removed and calculate source_bytes after the loop with audio_stream_frame_bytes() for any of the buffers, e.g. index 0?